### PR TITLE
Add to information model, auto generate schemas

### PR DESCRIPTION
**Pull request by lbradstreet**
This adds trigger events, and the event map to the information model and automatically generates schemas from them from the information model. Hopefully we will eventually do so with the rest of the schemas so we don't have any mismatch between what's in the schemas and what's in the information model.

### CODE REVIEW

File: src/onyx/schema.cljc [hunk 1]
```diff
@@ -620,3 +607,102 @@
...
+    :super-agg-fn (s/maybe Function)
+    (s/optional-key :new-window-state-fn) Function
+    (s/optional-key :grouping-fn) (s/cond-pre s/Keyword Function)}
+   record? 'record?))
+
+(def Event
+  (-> (information-model->schema (i/model :event-map))
+      (assoc (restricted-ns :onyx.core) s/Any)))
+
+(def WindowState
```
Comment (MichaelDrogalis) [inline src/onyx/schema.cljc]: Looks like this is declared twice.

Comment (lbradstreet) [inline src/onyx/schema.cljc]: Nice catch. My bad. Bad merge of course.

File: src/onyx/information_model.cljc [hunk 2]
```diff
@@ -507,6 +502,170 @@
              :optional? true
              :added "0.8.0"}}}
 
+   :event-map {:summary "Onyx exposes an 'event context' through many of its APIs. This is a description of what you will find in this map and what each of its key/value pairs mean. More keys
+may be added by the user as the context is associated to throughout the task pipeline."
+               :schema :onyx.schema.Event
+               :type :map
+               :model {:onyx.core/id {:type :uuid
```
Comment (MichaelDrogalis) [inline src/onyx/information_model.cljc]: Beautifullll work documenting each key here. Thank you!

File: doc/user-guide/information-model.md [hunk 3]
```diff
@@ -6,192 +6,4 @@ categories: [user-guide-page]
...
-|`:onyx.core/messenger-buffer` |`channel`  | The messenger buffer core.async channel for this peer|
-|`:onyx.core/messenger`        |`record`   | The Messenger record Component|
-|`:onyx.core/outbox-ch`        |`channel`  | The outbox core.async channel for this peer|
-|`:onyx.core/seal-ch`          |`channel`  | The core.async channel to deliver seal notifications for this job|
-|`:onyx.core/peer-opts`        |`map`      | The options that this peer was started with|
-|`:onyx.core/replica`          |`atom`     | The replica that this peer has currently accrued|
-|`:onyx.core/state`            |`atom`     | The state that this peer has accrued|
-|`:onyx.core/batch`            |`vector`   | The sequence of segments read by this peer|
-|`:onyx.core/results`          |`map`      | A map of read segment to a vector of segments produced by applying the function of this task|
+The information model is now described in the [Cheat Sheet](http://www.onyxplatform.org/docs/cheat-sheet/latest/). The information model is also described in a [nested map](https://github.com/onyx-platform/onyx/blob/master/src/onyx/information_model.cljc).
```
Comment (MichaelDrogalis) [inline doc/user-guide/information-model.md]: I was just getting ready to axe this. Good instincts.

**Comment by MichaelDrogalis** [comment]
Approved :)

**Pull request merged**